lua/TAair.lua

```diff
@@ -61,21 +62,74 @@ TAair = Class(TAunit)
         self:SetTurnMult(1)
 	end,
 
-	DoSelfDestruct = function(self)
-		self.DoSelfDestruct(self)
-		    if TAunit:BeenDestroyed() or TAunit.Dead then
-		        return
-		    end
-		
-		    if EntityCategoryContains(categories.TRANSPORTATION) then
-		        local cargo = TAunit:GetCargo()
-		        for k, v in cargo do
-		            if TAunit ~= v then
-		                OnKilled(v)
-		            end
-		        end
-			end
-		end,
 }
 
 TypeClass = TAair
+
+TAairTransport = Class(TAair, Unit) 

```

Comment:
I did say that I saw some examples of multiple inheritance in LUA.  But I did some tests, I'm not 100% it actually works when done this way ...